testr/testr.go

```diff
@@ -116,7 +116,7 @@ type testlogger struct {
 }
 
 func (l testlogger) GetUnderlying() *testing.T {
-	return l.t.(*testing.T)
+	return l.t.(*testing.T) //nolint:forcetypeassert
```

Comment:
That something like this would be needed was the main reason for having two separate implementations.

But I think it's okay and worth the reduction in code.

Reply:
I don't get it - if we can make this assumption, then what value is the interface-based version?

Reply:
At the API level, both versions are type-safe and only deal with `testing.T` resp. `TestingT`. The implementation of `testlogger` then reuses the interface-based version and therefore only has a `l.t` of type `TestingT` here although all that ever gets stored here is the `testing.T` pointer. Therefore the typecast is necessary and safe.

Reply:
I could call `NewWithInterface()` with something that ISN'T a `testing.T`, right?  Then this will panic.  What am I missing?

Reply:
This was added so that we could pass a `testing.B`

Reply:
`NewWithInterface()` returns a `newLoggerInterfaceWithOptions` and thus doesn't have this `testlogger.GetUnderlying`.

Only `New(t *testing.T)` and `NewWithOptions(t *testing.T, opts Options)` return a `testlogger` instance.

Reply:
I agree that this is non-obvious. Perhaps it is better to keep the two implementations completely separate, despite the code duplication? Or add more code comments?

Reply:
Ahhhhhh I see.  Yeah, a comment here like:

```
// This method is defined on testlogger, so the only type this could possibly be is testing.T